data/lua/wml/micro_ai.lua

```diff
@@ -19,11 +19,13 @@ function wesnoth.wml_actions.micro_ai(cfg)
     -- Check that the required common keys are all present and set correctly
     if (not cfg.ai_type) then wml.error("[micro_ai] is missing required ai_type= key") end
     if (not cfg.side) then wml.error("[micro_ai] is missing required side= key") end
-    if string.find(cfg.side, ',') then
-        local sides = stringx.split(cfg.side)
-        for _,side in ipairs(sides) do
-            cfg.side = tonumber(side)
-            wesnoth.wml_actions.micro_ai(cfg)
+    if string.find(cfg.side, ',') or string.find(cfg.side, '-') then
+        local side_ranges = stringx.split(cfg.side)
+        for _,side_range in ipairs(side_ranges) do
+            for side in stringx.iter_range(side_range) do
```

Comment:
There's also `iter_ranges` (note the S) that does the split for you.

Reply:
Oh, I now see what you meant. Wow, `iter_ranges` does all the parsing then. Thanks!